core/src/main/java/org/apache/iceberg/rest/requests/UpdateRequirementParser.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.iceberg.rest.requests;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.UncheckedIOException;
+import java.util.Locale;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.rest.requests.UpdateTableRequest.UpdateRequirement;
+import org.apache.iceberg.util.JsonUtil;
+
+public class UpdateRequirementParser {
+
+  private UpdateRequirementParser() {
+  }
+
+  private static final String TYPE = "type";
+
+  // assertion types
+  static final String ASSERT_TABLE_UUID = "assert-table-uuid";
+  static final String ASSERT_TABLE_DOES_NOT_EXIST = "assert-create";
+  static final String ASSERT_REF_SNAPSHOT_ID = "assert-ref-snapshot-id";
+  static final String ASSERT_LAST_ASSIGNED_FIELD_ID = "assert-last-assigned-field-id";
+  static final String ASSERT_CURRENT_SCHEMA_ID = "assert-current-schema-id";
+  static final String ASSERT_LAST_ASSIGNED_PARTITION_ID = "assert-last-assigned-partition-id";
+  static final String ASSERT_DEFAULT_SPEC_ID = "assert-default-spec-id";
+  static final String ASSERT_DEFAULT_SORT_ORDER_ID = "assert-default-write-order-id";
+
+  // AssertTableUUID
+  private static final String UUID = "uuid";
+
+  // AssertRefSnapshotID
+  // TODO - This is called `ref` in the spec.
+  //   https://github.com/apache/iceberg/blob/master/open-api/rest-catalog-open-api.yaml#L1359-L1360
+  private static final String NAME = "name";
```

Comment:
This field, `name`, is the name of the branch or tag. In the spec it's referred to as `ref`, but using `ref` could potentially cause confusion with `TableMetadata.ref` in this code here: https://github.com/apache/iceberg/blob/a19ac7835c8e251fd6a042e056f29ee63262adec/core/src/main/java/org/apache/iceberg/rest/requests/UpdateTableRequest.java#L261-L297

Reply:
If we made this `ref`, we'd just need to rename the current variable `ref` in the class's validate method to `baseRefType` or something: https://github.com/apache/iceberg/blob/a19ac7835c8e251fd6a042e056f29ee63262adec/core/src/main/java/org/apache/iceberg/rest/requests/UpdateTableRequest.java#L279-L281

Reply:
How about `ref-name`?

Reply:
I'm not sure that this actually needs to change. This parser is the only place where this name is used, so it doesn't really conflict or cause confusion if this is "ref". The `name` field of the assertion could still be set as "ref" in the serialization.

Reply:
Yeah. I'll just make the `NAME` field as `ref` in the parser.